token/cli/src/main.rs

```diff
@@ -1583,18 +1570,16 @@ async fn command_accounts(
     token: Option<Pubkey>,
     owner: Pubkey,
 ) -> CommandResult {
-    let program_id = if let Some(token) = token {
-        validate_mint(config, token).await?
-    } else {
-        config.program_id
-    };
+    if let Some(token) = token {
+        validate_mint(config, token).await?;
+    }
     let accounts = config
         .rpc_client
         .get_token_accounts_by_owner(
             &owner,
             match token {
                 Some(token) => TokenAccountsFilter::Mint(token),
-                None => TokenAccountsFilter::ProgramId(program_id),
+                None => TokenAccountsFilter::ProgramId(config.program_id),
```

Comment:
i left this as-is but dislike it. in my prettyprint pr i might make this ignore `-p` and print all accounts, grouped by program

Reply:
Yes please, that would be much better for usability